src/zenml/cli/stack.py

```diff
@@ -1273,3 +1275,115 @@ def copy_stack(
             )
         stack_wrapper.name = target_stack
         target_repo.zen_store.register_stack(stack_wrapper)
+
+
+@stack.command(
+    "register-secrets",
+    help="Interactively register all required secrets for a stack.",
+)
+@click.argument("stack_name", type=str, required=False)
+@click.option(
+    "--skip-existing",
+    "skip_existing",
+    is_flag=True,
+    default=False,
+    help="Skip secrets with existing values.",
+    type=bool,
+)
+def register_secrets(
+    skip_existing: bool,
+    stack_name: Optional[str] = None,
+) -> None:
+    """Interactively registers all required secrets for a stack.
+
+    Args:
+        skip_existing: If `True`, skip asking for secret values that already
+            exist.
+        stack_name: Name of the stack for which to register secrets. If empty,
+            the active stack will be used.
+    """
+    cli_utils.print_active_profile()
+
+    if stack_name:
+        try:
+            stack_ = Repository().get_stack(stack_name)
+        except KeyError:
+            cli_utils.error(f"No stack found for name `{stack_name}`.")
+    else:
+        stack_ = Repository().active_stack
+        cli_utils.declare(
+            f"No stack name specified, using the active stack `{stack_.name}`."
+        )
+
+    required_secrets = stack_.required_secrets
+    if not required_secrets:
+        cli_utils.declare("No secrets required for this stack.")
+        return
+
+    secrets_manager = stack_.secrets_manager
+    if not secrets_manager:
+        cli_utils.error(
+            "Unable to register secrets because the stack doesn't contain a "
+            "secrets manager."
```

Comment:
If it fails here, the user does not know if there are secrets required and so does not know if he should now register a secrets manager. Maybe we coul be explicit here: the following secrets would need to be reigstered: [] but there's no secrets manager, please register one like this